crates/core/src/binding.rs

```diff
@@ -546,34 +528,11 @@ impl<'a> Binding<'a> {
     /// Returns `None` if the binding is not bound to a list.
     pub(crate) fn list_items(&self) -> Option<impl Iterator<Item = NodeWithSource<'a>> + Clone> {
         match self {
-            Self::List(src, node, field_id) => {
-                let field_id = *field_id;
-                let mut cursor = node.walk();
-                cursor.goto_first_child();
-                let mut done = false;
-                Some(
-                    iter::from_fn(move || {
-                        // seems to me that the external loop is unnecessary, but was
-                        // getting an infinite loop without it.
-                        #[allow(clippy::never_loop)]
-                        while !done {
-                            while cursor.field_id() != Some(field_id) {
-                                if !cursor.goto_next_sibling() {
-                                    return None;
-                                }
-                            }
-                            let result = cursor.node();
-                            if !cursor.goto_next_sibling() {
-                                done = true;
-                            }
-                            return Some(result);
-                        }
-                        None
-                    })
-                    .filter(|child| child.is_named())
-                    .map(|named_child| NodeWithSource::new(named_child, src)),
-                )
-            }
+            Self::List(parent_node, field_id) => Some(
+                parent_node
+                    .children_by_field_id(*field_id)
+                    .filter(|item| item.node.is_named()),
```

Comment:
ideally `children_by_field_id` should just filter already. probably worth checking this doesn't break anything though

Reply:
I now have `children_by_field_id()` and `named_children_by_field_id()`. Using `named_children_by_field_id()` everywhere indeed resulted in some regressions.